Makefile

```diff
@@ -13,160 +13,76 @@
-# a pattern rule as follows:
-#
-#   %-up: %-build
-#       docker-compose -f $(DOCKER_COMPOSE_FILE) up -d --build $*
-#
-# The reason we are keeping the explicitly spelled-out versions is in order
-# to increase discoverability of the available Makefile targets and to enable
-# tab-completion of targets (which is not possible when using patterns).
-
-
-flowdb-up: flowdb-build
-	$(DOCKER_COMPOSE_UP) flowdb
-
-flowdb-down:
-	$(DOCKER_COMPOSE_DOWN) flowdb
-
-flowdb-build:
-	$(DOCKER_COMPOSE_BUILD) flowdb
-
-
-flowdb_testdata-up: flowdb_testdata-build
-	$(DOCKER_COMPOSE_UP) flowdb_testdata
-
-flowdb_testdata-down:
-	$(DOCKER_COMPOSE_DOWN) flowdb_testdata
-
-flowdb_testdata-build: flowdb-build
-	$(DOCKER_COMPOSE_BUILD) flowdb_testdata
-
-
-flowdb_synthetic_data-up: flowdb_synthetic_data-build
-	$(DOCKER_COMPOSE_UP) flowdb_synthetic_data
-
-flowdb_synthetic_data-down:
-	$(DOCKER_COMPOSE_DOWN) flowdb_synthetic_data
-
-flowdb_synthetic_data-build: flowdb-build
-	$(DOCKER_COMPOSE_BUILD) flowdb_synthetic_data
-
-
-flowmachine-up:
-	$(DOCKER_COMPOSE_UP) flowmachine
-
-flowmachine-down:
-	$(DOCKER_COMPOSE_DOWN) flowmachine
-
-flowmachine-build:
-	$(DOCKER_COMPOSE_BUILD) flowmachine
-
-
-flowapi-up:
-	$(DOCKER_COMPOSE_UP) flowapi
-
-flowapi-down:
-	$(DOCKER_COMPOSE_DOWN) flowapi
-
-flowapi-build:
-	$(DOCKER_COMPOSE_BUILD) flowapi
-
-
-flowauth-up:
-	$(DOCKER_COMPOSE_UP) flowauth
-
-flowauth-down:
-	$(DOCKER_COMPOSE_DOWN) flowauth
-
-flowauth-build:
-	$(DOCKER_COMPOSE_BUILD) flowauth
-
-
-worked_examples-up: worked_examples-build
-	$(DOCKER_COMPOSE_UP) worked_examples
-
-worked_examples-down:
-	$(DOCKER_COMPOSE_DOWN) worked_examples
+	$(DOCKER_COMPOSE) down
 
-worked_examples-build:
-	$(DOCKER_COMPOSE_BUILD) worked_examples
 
+# Per-service targets
+# While we could define these using a % pattern, those aren't tab-completable
 
-flowmachine_query_locker-up:
-	$(DOCKER_COMPOSE_UP_NOBUILD) flowmachine_query_locker
 
-flowmachine_query_locker-down:
-	$(DOCKER_COMPOSE_DOWN) flowmachine_query_locker
 
+flowdb_testdata-up flowdb_testdata-down: DOCKER_COMPOSE += -f $(DOCKER_COMPOSE_TESTDATA_FILE)
+flowdb_synthetic_data-up flowdb_synthetic_data-down : DOCKER_COMPOSE += -f $(DOCKER_COMPOSE_SYNTHETICDATA_FILE)
 
-flowetl-up:
-	$(DOCKER_COMPOSE_UP) flowetl
+%-build: DOCKER_COMPOSE += -f $(DOCKER_COMPOSE_FILE_BUILD)
 
-flowetl-down:
-	$(DOCKER_COMPOSE_DOWN) flowetl
+flowdb_testdata-build flowdb_synthetic_data-build: flowdb-build
 
-flowetl-build:
-	$(DOCKER_COMPOSE_BUILD) flowetl
+flowdb_testdata-build : DOCKER_COMPOSE += -f $(DOCKER_COMPOSE_TESTDATA_FILE_BUILD)
+flowdb_synthetic_data-build : DOCKER_COMPOSE += -f $(DOCKER_COMPOSE_SYNTHETICDATA_FILE_BUILD)
 
 
-flowetl_db-up:
-	$(DOCKER_COMPOSE_UP) flowetl_db
+$(services:=-up):
+	$(DOCKER_COMPOSE) up -d $(subst _synthetic_data,,$(subst _testdata,,$(@:-up=)))
 
-flowetl_db-down:
-	$(DOCKER_COMPOSE_DOWN) flowetl_db
+$(services:=-down):
+	$(DOCKER_COMPOSE) down $(subst _synthetic_data,,$(subst _testdata,,$(@:-down=)))
```

Comment:
Previously `make <service>-down` did `docker-compose rm -f -s -v <service>`. Again, not sure if this is a deliberate change, but `docker-compose down <service>` is invalid usage so this target doesn't work at the moment.